src/actions/storage-actions.js

```diff
@@ -43,8 +43,9 @@ export const getDevicesAction = () => {
     return async (dispatch) => {
         try {
             const devices = await getDevices();
+            const deviceData = {};
             const mountPoints = await getMountPoints();
-            const devicesData = await Promise.all(devices.map(async (device) => {
+            for (const device of devices) {
```

Comment:
Just reading loud, so this will also gather the data serially instead of concurrently, which is OK (perhaps even desirable to be safer) I think.

Reply:
Yes, honestly I did not serialize this on the purpose. I was trying to clean up the code and make it a bit more readable. I guess it's ok side effect, we dont have performance issues here.